src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    internal class EventHubScaleMonitor : IScaleMonitor<EventHubTriggerMetrics>
+    {
+        private const string EventHubContainerName = "azure-webjobs-eventhub";
+        private const int PartitionLogIntervalInMinutes = 5;
+
+        private readonly string _functionId;
+        private readonly string _eventHubName;
+        private readonly string _consumerGroup;
+        private readonly EventHubsConnectionStringBuilder _connectionStringBuilder;
+        private readonly CloudBlobContainer _blobContainer;
+        private readonly ScaleMonitorDescriptor _scaleMonitorDescriptor;
+        private readonly ILogger _logger;
+
+        private EventHubClient _client;
+        private DateTime _nextPartitionLogTime;
+        private DateTime _nextPartitionWarningTime;
+
+        public EventHubScaleMonitor(
+            string functionId,
+            string eventHubName,
+            string consumerGroup,
+            string connectionString,
+            string storageConnectionString,
+            ILogger logger,
+            CloudBlobContainer blobContainer = null)
+        {
+            _functionId = functionId;
+            _eventHubName = eventHubName;
+            _consumerGroup = consumerGroup;
+            _logger = logger;
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{_functionId}-EventHubTrigger-{_eventHubName}".ToLower());
+            _nextPartitionLogTime = DateTime.UtcNow;
+            _nextPartitionWarningTime = DateTime.UtcNow;
+
+            _connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString);
+            if (!String.IsNullOrEmpty(_eventHubName))
+            {
+                _connectionStringBuilder.EntityPath = _eventHubName;
+            }
+            _client = EventHubClient.CreateFromConnectionString(_connectionStringBuilder.ToString());
+
+            _blobContainer = blobContainer;
+            if (_blobContainer == null)
+            {
+                CloudStorageAccount cloudStorageAccount = CloudStorageAccount.Parse(storageConnectionString);
+                CloudBlobClient blobClient = cloudStorageAccount.CreateCloudBlobClient();
+                _blobContainer = blobClient.GetContainerReference(EventHubContainerName);

```

Comment:
These will throw exceptions if the connection string is incorrect or the account doesn't exist. I assumed we would want those exceptions because the ScaleMonitor wouldn't be able to do its job without these, but I wasn't sure how that might affect the host

Reply:
The connection string is the default AzureWebJobsStorage connection string for the app, right? These exceptions will be thrown when the host tries to access the monitor as part of host startup [here](https://github.com/Azure/azure-webjobs-sdk/blob/2c7b2f30b4d197d091f5aa5543ddf65d6a34e689/src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs#L103) and that will be a fatal exception (host won't start, but the error will be logged). However, if the default storage connection string is incorrect, I suspect the host will blow up in a similar way anyways, even if your monitor wasn't registered. Can you test this out? Compare the host behavior with an invalid connection string with/without your monitor. I think we're good though.

Reply:
Tested both scenarios -- both dev and my changes will throw the same exception when the EPH tries to connect to the storage account.